core/primitives/src/block.rs

```diff
@@ -315,6 +313,14 @@ impl Block {
         }
     }
 
+    pub fn compute_tx_root(transactions: &Vec<SignedTransaction>) -> CryptoHash {
+        merklize(
+            // TODO What exactly I should take from transactions to build a tx_root? sort somehow?
+            &transactions.iter().map(|tx| tx.transaction.nonce).collect::<Vec<Nonce>>(),
```

Comment:
`SignedTransaction::get_hash()`?

nonce is naturally incorrect, because transactions that originate from different originators can have the same nonce.
No need to sort, `transactions` is already an ordered set.